charts/opentelemetry-collector/templates/daemonset.yaml

```diff
@@ -10,6 +10,17 @@ spec:
     matchLabels:
       {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
       component: agent-collector
+  updateStrategy:
+    rollingUpdate:
+      {{- if .Values.agentCollector.rollingUpdate.maxSurge }}
+      maxSurge: {{ .Values.agentCollector.rollingUpdate.maxSurge }}
+      {{- end }}
+      {{- if .Values.agentCollector.rollingUpdate.maxUnavailable }}
+      maxUnavailable: {{ .Values.agentCollector.rollingUpdate.maxUnavailable }}
+      {{- else }}
+      maxUnavailable: 1
+      {{- end }}
+    type: {{ .Values.agentCollector.updateStrategy }}
```

Comment:
If someone set `.Values.agentCollector.updateStrategy` to `OnDelete`, what would the above `rollingUpdate` section do?  My gut says that we should only render a `rollingUpdate` section if the `updateStrategy` is set to `RollingUpdate`.